src/uct/sm/self/self.c

```diff
@@ -0,0 +1,363 @@
+                                   UCT_IFACE_FLAG_AM_SHORT         |
+                                   UCT_IFACE_FLAG_AM_BCOPY         |
+                                   UCT_IFACE_FLAG_PUT_SHORT        |
+                                   UCT_IFACE_FLAG_PUT_BCOPY        |
+                                   UCT_IFACE_FLAG_GET_BCOPY        |
+                                   UCT_IFACE_FLAG_ATOMIC_ADD32     |
+                                   UCT_IFACE_FLAG_ATOMIC_ADD64     |
+                                   UCT_IFACE_FLAG_ATOMIC_FADD64    |
+                                   UCT_IFACE_FLAG_ATOMIC_FADD32    |
+                                   UCT_IFACE_FLAG_ATOMIC_SWAP64    |
+                                   UCT_IFACE_FLAG_ATOMIC_SWAP32    |
+                                   UCT_IFACE_FLAG_ATOMIC_CSWAP64   |
+                                   UCT_IFACE_FLAG_ATOMIC_CSWAP32   |
+                                   UCT_IFACE_FLAG_ATOMIC_CPU       |
+                                   UCT_IFACE_FLAG_PENDING          |
+                                   UCT_IFACE_FLAG_CB_SYNC          |
+                                   UCT_IFACE_FLAG_EP_CHECK;
+
+    attr->cap.put.max_short       = UINT_MAX;
+    attr->cap.put.max_bcopy       = SIZE_MAX;
+    attr->cap.put.min_zcopy       = 0;
+    attr->cap.put.max_zcopy       = 0;
+    attr->cap.put.opt_zcopy_align = 1;
+    attr->cap.put.align_mtu       = attr->cap.put.opt_zcopy_align;
+    attr->cap.put.max_iov         = 1;
+
+    attr->cap.get.max_bcopy       = SIZE_MAX;
+    attr->cap.get.min_zcopy       = 0;
+    attr->cap.get.max_zcopy       = 0;
+    attr->cap.get.opt_zcopy_align = 1;
+    attr->cap.get.align_mtu       = attr->cap.get.opt_zcopy_align;
+    attr->cap.get.max_iov         = 1;
+
+    attr->cap.am.max_short        = iface->send_size;
+    attr->cap.am.max_bcopy        = iface->send_size;
+    attr->cap.am.min_zcopy        = 0;
+    attr->cap.am.max_zcopy        = 0;
+    attr->cap.am.opt_zcopy_align  = 1;
+    attr->cap.am.align_mtu        = attr->cap.am.opt_zcopy_align;
+    attr->cap.am.max_hdr          = 0;
+    attr->cap.am.max_iov          = 1;
+
+    attr->latency.overhead        = 0;
+    attr->latency.growth          = 0;
+    attr->bandwidth               = 6911 * 1024.0 * 1024.0;
+    attr->overhead                = 10e-9;
+    attr->priority                = 0;
+
+    return UCS_OK;
+}
+
+static ucs_status_t uct_self_iface_get_address(uct_iface_h tl_iface,
+                                               uct_iface_addr_t *addr)
+{
+    const uct_self_iface_t *iface = ucs_derived_of(tl_iface, uct_self_iface_t);
+
+    *(uct_self_iface_addr_t*)addr = iface->id;
+    return UCS_OK;
+}
+
+static int uct_self_iface_is_reachable(const uct_iface_h iface, const uct_device_addr_t *dev_addr,
+                                       const uct_iface_addr_t *iface_addr)
+{
+    const uct_self_iface_t *self_iface = NULL;
```

Comment:
I see that code is just copied from another file, but why we initialize vars here? 3 lanes below we re-init it...
and alignment is not ok